aiida/repository/repository.py

```diff
@@ -378,7 +372,10 @@ def open(self, path: FilePath) -> io.BufferedReader:
         :raises IsADirectoryError: if the object is a directory and not a file.
         :raises OSError: if the file could not be opened.
         """
-        with self.backend.open(self.get_file(path).key) as handle:
+        key = self.get_file(path).key
+        if key is None:
+            raise TypeError('File key not set')
```

Comment:
So I see why mypy thinks it needs this, because technically `File.key` can indeed return `str` or `None`. However, if the `File` has `file_type==FileType.File` the `key` cannot be `None` because this is checked in the constructor. So we should never get in this `TypeErrorr` since if `key is None` than `get_file` will already have raised `IsADirectoryError` because it is not a file. So I am wondering if this should not simply be an `AssertionError` if we really need to have this check here, because it would be an internal coding inconsistency.

Reply:
done 👍 
yeh this is why I personally probably would have erred towards having File and Directory subclasses, but appreciate it is somewhat a matter of personal preference

Reply:
There might be a problem on my side, but I still see the `TypeError`. Did you change this to
```
assert self.get_file(path).key is not None
```
?

Reply:
Oh yeh, I changed it in `delete_object`, but not here

Reply:
done